test/parallel_api/numeric/numeric.ops/reduce_by_segment.pass.cpp

```diff
@@ -327,10 +327,10 @@ main()
         test4buffers<sycl::usm::alloc::device, test_reduce_by_segment<ValueType, BinaryPredicate, BinaryOperation>>();
 #endif // TEST_DPCPP_BACKEND_PRESENT
 
-#if !_PSTL_ICC_TEST_SIMD_UDS_BROKEN
+#if !_PSTL_ICC_TEST_SIMD_UDS_BROKEN && !_PSTL_ICPX_TEST_RED_BY_SEG_OPTIMIZER_CRASH
```

Comment:
I think it is better to have two macros, because the issues have different root cases and environments to reproduce.

For example, if we drop support of the compiler when `_PSTL_ICC_TEST_SIMD_UDS_BROKEN` is effective, we would want to get rid of that macro and thus to decouple that logic.

Reply:
I agree with @dmitriy-sobolev with regards to this. Additionally, since `_PSTL_ICC_TEST_SIMD_UDS_BROKEN` is a broader compiler issue I believe, there may be a need to add it elsewhere in the future. I think keeping these two separate is best.